xbmc/windowing/WinSystem.h

```diff
@@ -260,6 +260,24 @@ class CWinSystemBase
    */
   std::pair<bool, int> GetDitherSettings();
 
+  /*!
+   * \brief Binds a shared context to the current thread, in order to upload textures asynchronously.
+   * \return Return true if a texture upload context exists and the binding succeeds.
+   */
+  virtual bool BindTextureUploadContext() { return false; }
```

Comment:
I've experimented with that, but I couldn't come up with an universal solution which does not look hacky. If you could give me some pointers, I'd appreciate it.

Reply:
See here: https://github.com/xbmc/xbmc/blob/64e0ecf2f3d0a6011f97067f0922605bc14e453b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIMEGLES.cpp#L59

The problem is that x11 uses a different EGL context class so that might be annoying. It was always the intention to get x11 to use the same EGL class that wayland and gbm are using, it just hasn't happened yet.

Reply:
Yeah, that's why I shelved the branch for a long time. I'm still running X11. I'll see what I can do.

On the flip side, I believe that GLX and DX11 can have a very similar implementations. It is not EGL specific, it that's your concern.